google/data_source_google_service_account_id_token_test.go

```diff
@@ -0,0 +1,73 @@
+	"client_email": "user@gcp-project.iam.gserviceaccount.com",
+	"client_id": "103198861025845558729",
+	"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+	"token_uri": "https://accounts.google.com/o/oauth2/token",
+	"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+	"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/user%40gcp-project.iam.gserviceaccount.com"
+  }
+  `
+
+const targetAudience = "https://foo.bar/"
+const fakeIdToken = "eyJhbGciOiJSUzI1NiIsIm..."
+
+func testAccCheckServiceAccountIdTokenValue(name, value string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Outputs[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		// TODO: validate the token belongs to the service account
+		if rs.Value == "" {
+			return fmt.Errorf("%s Cannot be empty", name)
+		}
+
+		return nil
+	}
+}
+
+func TestAccDataSourceGoogleServiceAccountIdToken_basic(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "data.google_service_account_id_token.default"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleServiceAccountIdToken_datasource(targetAudience),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "target_audience", targetAudience),
+					testAccCheckServiceAccountIdTokenValue("google_service_account_id_token.default", fakeIdToken),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleServiceAccountIdToken_datasource(targetAudience string) string {
```

Comment:
I think it's also worth having a test that sets more of the available fields in the data source. It also looks from the way the docs and resource are structured that there's two distinct use cases for this data source. Can you make sure both use cases are tested?

Reply:
addin basic impersonation and basic/regular usage testcases.